pyxnat/core/resources.py

```diff
@@ -1593,6 +1817,27 @@ def params(self):
 @add_metaclass(ElementType)
 class Resource(EObject):
 
+    def __repr__(self):
+        resource_id = uri_last(self._uri)
+
+        # Check if resource exists
+
+        if self.exists():
+
+            # Fetch data files
+            files_count = len(self.files().fetchall())
```

Comment:
Not an straight-forward task but we could perhaps also consider including other attributes as well, e.g. `file_size`. See an example list of available attributes per resource object:
```
{'xnat_abstractresource_id': '14130',
 'label': 'SPM12_SEGMENT_T2T1_COREG7',
 'element_name': 'xnat:resourceCatalog',
 'category': 'resources',
 'cat_id': 'XNAT_E00001',
 'cat_desc': ' ',
 'file_count': '17',
 'file_size': '334253120',
 'tags': '',
 'content': '',
 'format': ''}
```

EDIT: I also noticed that this command below takes a decent amount of time to execute, specially for resources with a large number of files. That might be an additional argument for going the way suggested above.
```
# Fetch data files
files_count = len(self.files().fetchall())
```

Reply:
Could be a good idea indeed. And what about using the `file_count` attribute?
I don't remember how I got to use `fetchall()`. I just would like to make sure that these `file_count` and `file_size` fields always exist, not like those we found a few days ago (`ses`, etc.). What do you think?

Reply:
I will check if both fields always exists and if that is the case we get both attributes from there.

Reply:
Great. 

Yes, @xgrg, that's my idea as well to include both, see the comment's EDIT. Basically the problem I found is that I can't get those elements with an elegant (`KeyError` exception), 
```
resource.attrs.mget(['file_count', 'file_size'])
```
The XNAT REST API command `[GET] '/data/experiments/EXP/resources&format=FORMAT'` works fine however. Seems like we found a new issue?

As a pragmatic approach, we could do something as follows:
```
def __repr__(self):
        resource_id = self.id()

        # Check if resource exists

        if self.exists():

            # Fetch data files
            base_url = uri_parent(self._uri)
            resources = self._intf._get_json(base_url)
            res_info = [r for r in resources 
                             if r['xnat_abstractresource_id'] == resource_id][0]

            # Creating the output string to be returned
            output = '<{cl} Object> {id} `{label}` ({fc} files {fs} MB)'
            output = output.format(label=self.label(),
                                   cl=self.__class__.__name__,
                                   id=resource_id,
                                   fc=res_info['file_count'],
                                   fs=float(res_info['file_size'])/1024*2)
            return output
        else:
            return '<%s Object> %s' % (self.__class__.__name__,
                                       unquote(uri_last(self._uri)))
```